src/stream/src/executor/mview/materialize.rs

```diff
@@ -699,10 +703,43 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
             lru_cache,
             row_serde,
             version_column_indices,
+            vnode_max_keys: BTreeMap::new(),
             _serde: PhantomData,
         }
     }
 
+    async fn init_vnode_max_keys<S: StateStore>(
+        &mut self,
+        state_table: &StateTableInner<S, SD>,
+    ) -> StreamExecutorResult<()> {
+        assert!(self.vnode_max_keys.is_empty());
+        'vnode_loop: for vnode in state_table.vnodes().iter_vnodes() {
+            let table_iter = state_table
+                .rev_iter_keyed_row_with_vnode(
+                    vnode,
+                    &(Bound::<OwnedRow>::Unbounded, Bound::<OwnedRow>::Unbounded),
+                    PrefetchOptions::new(false, false),
```

Comment:
nits: This is not the subject of this PR. **PrefetchOptions** should be an Optional type. PrefetchOptions::new(false, false) is not intuitive, as it means prefetching is disabled. Alternatively, we could construct it via a dedicated function, such as Prefetch::without_prefetch()....